### ISSUE TITLE
osm controller: Error fetching certificate manager of kind tresor

### ISSUE THREAD

**Issue by ritazh** [openservicemesh/osm#1829]
**Bug description**:
After deploying via `helm install` from the latest release v0.4.1, getting pod CrashLoopBackOff and this error:
```console
{"level":"fatal","component":"kube-events","error":"Invalid secret for certificate","reason":"FatalInvalidCertificateManager","time":"2020-10-13T20:07:03Z","file":"events.go:139","message":"Error fetching certificate manager of kind tresor"}
```

**Affected area (please mark with X where applicable)**:
- Install                    [X]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [X]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- CI System                  [ ]

**Expected behavior**:
Deployment is successful

**Steps to reproduce the bug (as precisely as possible)**:

**How was OSM installed?**:
`helm install` from the latest release v0.4.1

**Anything else we need to know?**:

**Environment**:
- OSM version (use `osm version`): chart version: 0.4.1,  osm image tag: v0.4.1
- Kubernetes version (use `kubectl version`): v1.18.0 and v1.19.0
- Size of cluster (number of worker nodes in the cluster): kind cluster
- Others:

**Comment by ritazh** [comment]
cc @nojnhuh can you PTAL?

**Comment by ritazh** [comment]
Update on this:

1. Repro'ed this issue with `helm install` 0.4.1 on a fresh cluster. 
2. `helm install` latest chart form https://github.com/openservicemesh/osm/commit/c1265f507c5d488f2058fe97667a97cc21823e86 off main, this bug only happens on a reinstall of the chart but does not occur on a fresh cluster.

**Comment by shashankram** [comment]
@ritazh I see why this issue is seen on release v0.4.1 and main.

On v0.4.1:
- missing change https://github.com/openservicemesh/osm/pull/1816

On main:
- Upon Helm install and delete, the issue is seen because the default controller image `openservicemesh/osm-controller:v0.4.1` does not have the change https://github.com/openservicemesh/osm/pull/1816


We can cut v0.4.2 with this fix, updating the default image should resolve the issue you are seeing.

**Comment by shashankram** [comment]
Resolved in release v0.4.2 https://github.com/openservicemesh/osm/releases/tag/v0.4.2
